qualibration_libs/analysis/models.py

```diff
@@ -182,3 +190,224 @@ def decay_exp(t, a, offset, decay, **kwargs):
 #     return (A + k * w) * (
 #         1 - ((Q / Qe) / (1 + 2 * 1j * Q * (w - omega_r) / (omega_0 + omega_r)))
 #     )
+
+class S21Resonator:
+    """
+    A class to fit 1D resonator spectroscopy data using the S21 model.
+    Ref: https://doi.org/10.7907/RAT0-VM75
+    """
+    _DELAY_FIT_PERCENTAGE = 10
+    _PHASES_THRESHOLD_PERCENTAGE = 80
+    _STD_DEV_GAUSSIAN_KERNEL = 5
+    _PHASE_ELEMENTS = 5
```

Comment:
Same comment here, are the users supposed to play with these parameters, and if so, how can they know how they impact the fit.
For instance on dataset #544, qC2 is systematically failing, but it is very hard to understand why.